Templates/Common-Backend-Scripts/utilities/fetchBuildDependencies.groovy

```diff
@@ -0,0 +1,339 @@
+@groovy.transform.BaseScript com.ibm.dbb.groovy.ScriptLoader baseScript
+import com.ibm.dbb.metadata.*
+import com.ibm.dbb.dependency.*
+import com.ibm.dbb.build.*
+import groovy.transform.*
+import com.ibm.dbb.build.report.*
+import com.ibm.dbb.build.report.records.*
+import groovy.yaml.YamlSlurper
+import groovy.yaml.YamlBuilder
+import groovy.lang.GroovyShell
+import groovy.cli.commons.*
+import groovy.util.*
+import java.nio.file.*
+
+
+// script properties
+@Field Properties props = new Properties()
+@Field def artifactRepositoryHelpers
+
+// Parse arguments from command-line
+parseArgs(args)
+
+// Print parms
+println("** Script configuration:")
+props.each { k,v->
+	println "   $k -> $v"
+}
+
+File artifactRepositoryHelpersScriptFile = new File(props.artifactRepositoryHelpersScript)
+if (artifactRepositoryHelpersScriptFile.exists()) {
+	artifactRepositoryHelpers = loadScript(artifactRepositoryHelpersScriptFile)
+} else {
+	println("*! [ERROR] The Artifact Repository Helper script '${props.artifactRepositoryHelpersScript}' doesn't exist. Exiting.")
+	System.exit(1)
+}
+
+File applicationDescriptorFile = new File(props.applicationDescriptor)
+if (!applicationDescriptorFile.exists()) {
+	println("*! [ERROR] The Application Descriptor file '${props.applicationDescriptor}' doesn't exist. Exiting.")
+	System.exit(1)
+}
+
+// setup import directory
+File importFolder = new File("$props.workspace/imports")
+if (importFolder.exists()) importFolder.deleteDir()
+importFolder.mkdirs()
+
+// setup package cache
+cacheFolder = new File("/tmp/.pkg")
```

Comment:
Can we externalize this cache folder location with a property in the pipelineBackend.config?

Reply:
I have updated the pipelineBackend.config with new settings and implemented the logic for the fetch